### PR TITLE
sphinx doc skeleton

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

Here we add the skeleton for sphinx docs, as well as the github workflow. 

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: .github/workflows/docs.yml
Comment (Yurlungur) [inline .github/workflows/docs.yml]: This should be set to `main` only right before final merge.

File: .github/workflows/docs.yml
Comment (Yurlungur) [inline .github/workflows/docs.yml]: Make this change right before merge:
```suggestion
        - name: Deploy
          if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
          uses: peaceiris/actions-gh-pages@v3.7.3
          with:
            github_token: ${{ secrets.GITHUB_TOKEN }}
            publish_dir: ./doc/sphinx/_build/html
            force_orphan: true
```

File: doc/index.html
Comment (Yurlungur) [inline doc/index.html]: Change right before merge:
```suggestion
   <link rel="canonical" href="https://parthenon-hpc-lab.github.io/parthenon/develop/index.html">
```

File: doc/index.html
Comment (Yurlungur) [inline doc/index.html]: Change before merge
```suggestion
    <meta http-equiv="refresh" content="0; url=./develop/index.html">
```

**Comment by Yurlungur** [comment]
Docs are here. After merge we need to commit my above suggestions to make it redirect to develop branch:
https://parthenon-hpc-lab.github.io/parthenon/jmm/sphinx-doc-skeleton/index.html

File: .github/workflows/docs.yml
Comment (Yurlungur) [inline .github/workflows/docs.yml]:
```suggestion
    branches: [ main ]
```

**Review by c-prather — APPROVED** [review]
Does the thing. Hopefully still does the thing after branch name changes.  There might be an issue of 'main' vs 'develop' branch naming in some of the proposed changes?

File: .github/workflows/docs.yml
Comment (pgrete) [reply .github/workflows/docs.yml]: Should this be `develop` rather than `main` or does this not matter as `develop` is our default (`main`) branch?

File: doc/index.html
Comment (pgrete) [reply doc/index.html]: Probably also in the line above, doesn't it?

File: doc/sphinx/conf.py
Comment (pgrete) [inline doc/sphinx/conf.py]:
```suggestion
copyright = "2023, The Parthenon Collaboration and Triad National Security"
```

**Review by pgrete — COMMENTED** [review]
Does the current setup automatically generate docs for all (new, following this PR) branches or only for `develop` (i.e., after a merge).
I wouldn't mind if we build docs by default for all branches.

**Comment by Yurlungur** [comment]
> There might be an issue of 'main' vs 'develop' branch naming in some of the proposed changes?

Oh good catch. I copied this from another project. I'll fix that.

**Comment by Yurlungur** [comment]
> Does the current setup automatically generate docs for all (new, following this PR) branches or only for `develop` (i.e., after a merge).
> I wouldn't mind if we build docs by default for all branches.

Should build automatically for any branch with an open PR, as well as release tags,  which I think is a good balance. You can navigate to the bottom left of the sphinx page to select which branch/tag you are looking at

File: .github/workflows/docs.yml
Comment (pgrete) [inline .github/workflows/docs.yml]:
```suggestion
    branches: [ develop ]
```

File: .github/workflows/docs.yml
Comment (pgrete) [inline .github/workflows/docs.yml]:
```suggestion
    branches: [ develop ]
```

File: doc/index.html
Comment (pgrete) [inline doc/index.html]:
```suggestion
    <meta http-equiv="refresh" content="0; url=./develop/index.html">
    <link rel="canonical" href="https://parthenon-hpc-lab.github.io/parthenon/develop/index.html">
```

**Review by pgrete — APPROVED** [review]

**Comment by pgrete** [comment]
> > There might be an issue of 'main' vs 'develop' branch naming in some of the proposed changes?
> 
> Oh good catch. I copied this from another project. I'll fix that.

I didn't see the fixes (maybe not pushed?) so I now quickly made them myself.
Hitting auto-merge now.

**Comment by pgrete** [comment]
@Yurlungur do you know why it built successfully but didn't deploy? https://github.com/parthenon-hpc-lab/parthenon/actions/runs/3938060249/jobs/6736303606
-> 404 not found: https://parthenon-hpc-lab.github.io/parthenon/develop/index.html

**Comment by Yurlungur** [comment]
I think I know. This line:
```
if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
```
prevents deployment from forks, which is needed because the action only has push permissions from our repo and so it only works on branches. However, it also implies it only works for pull requests, not merges into develop and not nightly builds.

That's why it deployed for, e.g., @brryan 's most recent PR, and why the the documentation on develop is now present.

I *think* the following PR should fix it: #814.

**Pull request merged**
